### PR TITLE
Misalignment updates FST and Simplified FST Geometry

### PR DESCRIPTION
**Pull request by klendathu2k**
Updates to misalignment tables (names, correct timestamps, etc...) for the FST.  Implementation of simplified FST geometry model for track reconstruction.  Support for misaligned geometries in g2t_volume_id.

### CODE REVIEW

**Review by genevb — COMMENTED** [review]
I don't have much to speak of on any of this PR (I'm sure misalignment will be useful). A tangential question though: what else is needed for `y2022a`?

**Comment by klendathu2k** [comment]
Depends on whether forward tracking is to be supported with y2022a.  If 
we limit
ourselves to support central tracking with the y2022a tag, I can define 
that tomorrow.
Otherwise, I have to double check where we are on the sTGC.

**Review by akioogawa — APPROVED** [review]

**Comment by genevb** [comment]
> Depends on whether forward tracking is to be supported with y2022a. If we limit ourselves to support central tracking with the y2022a tag, I can define that tomorrow. 

If you define y2022a now (sans forward tracking), is there any difference from y2022 (initial)?

**Comment by klendathu2k** [comment]
There should not be.  It will be a straight y2022 --> y2022a copy.  
There have
been no updates outside of the forward detectors.

**Comment by genevb** [comment]
> There should not be. It will be a straight y2022 --> y2022a copy. There have been no updates outside of the forward detectors.

Thanks for the reply. In that case, I am not pressing for freezing y2022a now.

**Comment by plexoos** [comment]
Should we store the misalignment values from StarDb/Geometry/fst/*.C in the database instead?

**Comment by klendathu2k** [comment]
@plexoos The misalignments are not expected to vary over the course of the run, so we only expect one entry once they have been measured.   The practice for HFT misalignments, and I believe TPC and others as well, has been to store them StarDb area.

**Comment by klendathu2k** [comment]
@genevb @perevbnlgov  This geometry update is stalled waiting for a code owner to approve.  Could one of you have a look?  Thanks.

File: StRoot/StChain/GeometryDbAliases.h
Comment (genevb) [inline StRoot/StChain/GeometryDbAliases.h:162]: Why not use 20211015, as is the case for y2022, for all the dev2022* timestamps?

File: StRoot/StChain/GeometryDbAliases.h
Comment (genevb) [inline StRoot/StChain/GeometryDbAliases.h]: Can the comments here indicate the difference between `dev2022`,` dev2022m`, and `dev2022sm`?

**Comment by genevb** [comment]
> The misalignments are not expected to vary over the course of the run, so we only expect one entry once they have been measured. The practice for HFT misalignments, and I believe TPC and others as well, has been to store them StarDb area.

For the TPC, I think surveys and ideal geometries have been stored as CINT files, but the misalignments used in production are stored in the database. If my memory serves me correctly, I think for the HFT that the CINT files were generated from the database tables, but I could be wrong.

-Gene

File: StRoot/StChain/GeometryDbAliases.h
Comment (klendathu2k) [reply StRoot/StChain/GeometryDbAliases.h:162]: I have been trying to keep the R&D / development geometries well advanced in time, so as not to accidentally run them in a production context (thus the +100 years).

File: StRoot/StChain/GeometryDbAliases.h
Comment (klendathu2k) [reply StRoot/StChain/GeometryDbAliases.h]: @genevb Comments updated.  Any other issues?

**Comment by klendathu2k** [comment]
Looks like trigger simulator is breaking tests?

**Comment by genevb** [comment]
> Looks like trigger simulator is breaking tests?

I have created an issue: #303 .

-Gene

**Comment by veprbl** [comment]
FYI I've restarted CI after #304 was merged, the tests are green again

**Comment by starsdong** [comment]
Are we good with this one? @jdbrice or @perevbnlgov need your approval to proceed.

**Review by plexoos — APPROVED** [review]
I trust this been tested and ready

**Comment by starsdong** [comment]
Dmitri, let us move ahead and merge this in then. Thanks

**Pull request merged**
